src/compiler/checker.ts

```diff
@@ -14612,8 +14612,15 @@ namespace ts {
         function checkUnusedTypeParameters(node: ClassDeclaration | ClassExpression | FunctionDeclaration | MethodDeclaration | FunctionExpression | ArrowFunction | ConstructorDeclaration | SignatureDeclaration | InterfaceDeclaration) {
             if (compilerOptions.noUnusedLocals && !isInAmbientContext(node)) {
                 if (node.typeParameters) {
+                    // Only report errors on the last declaration for the type parameter container;
+                    // this ensures that all uses have been accounted for.
+                    const symbol = node.symbol && getMergedSymbol(node.symbol);

```

Comment:
Can you use `getSymbolOfNode(node)` here instead? `getMergedSymbol` already checks that `node.symbol` is defined.